pkg/user/user_integration_test.go

```diff
@@ -41,63 +44,145 @@ func TestUserHandler(t *testing.T) {
 	authorization := middleware.NewAuthorization(slog.New(slog.NewTextHandler(os.Stdout, nil)), userService)
 	key, err := rsa.GenerateKey(rand.Reader, 2048)
 	require.NoError(t, err, "failed to generate private key")
-	// TODO(DEVOPS-259) we should not use a pointer as we do not mutate and should not mutate the
-	// certificate
+	// TODO(DEVOPS-259) we should not use a pointer as we do not mutate and should not mutate the certificate
 	authentication := middleware.NewAuthentication(&key.PublicKey, userService)
 
 	logger := slog.New(slog.NewTextHandler(os.Stdout, nil))
 	redis := inttest.SetupRedis(t)
 	tokenRepository := token.NewRepository(redis)
-	tokenService, err := token.NewService(logger, tokenRepository, key, &key.PublicKey, 10, "secret", 10)
+	tokenService, err := token.NewService(logger, tokenRepository, key, &key.PublicKey, 10, "secret", 20, 30)
 	require.NoError(t, err)
 
 	client := inttest.SetupHTTPServer(t, func(engine *gin.Engine) {
-		userHandler := user.NewHandler("hostname", config.Authentication{}, userService, tokenService)
+		userHandler := user.NewHandler("hostname", 10, 20, 30, &key.PublicKey, userService, tokenService)
 		user.Routes(engine, authentication, authorization, userHandler)
 	})
 
-	var user1ID string
-	{
-		t.Log("SignUpUsers")
+	t.Run("SignUp", func(t *testing.T) {
+		t.Parallel()
+
+		t.Log("SignUpUser")
 
-		var user1 model.User
+		var user model.User
 		client.PostJSON(t, "/users", strings.NewReader(`{
-			"email":    "user1@dhis2.org",
+			"email":    "user@dhis2.org",
 			"password": "oneoneoneoneoneoneone111"
-		}`), &user1)
+		}`), &user)
 
-		require.Equal(t, "user1@dhis2.org", user1.Email)
-		require.Empty(t, user1.Password)
-		user1ID = strconv.FormatUint(uint64(user1.ID), 10)
+		assert.Equal(t, "user@dhis2.org", user.Email)
+		assert.Empty(t, user.Password)
+		assert.Empty(t, user.EmailToken)
+		assert.False(t, user.Validated)
 
-		u1, err := userService.FindById(context.Background(), user1.ID)
-		require.NoError(t, err)
-		err = userService.ValidateEmail(u1.EmailToken)
+		t.Log("ValidateEmail")
+
+		u, err := userService.FindById(context.Background(), user.ID)
 		require.NoError(t, err)
+		requestBody := strings.NewReader(`{"token": "` + u.EmailToken.String() + `"}`)
+		client.Do(t, http.MethodPost, "/users/validate", requestBody, http.StatusOK, inttest.WithHeader("Content-Type", "application/json"))
 
-		var user2 model.User
-		client.PostJSON(t, "/users", strings.NewReader(`{
-			"email":    "user2@dhis2.org",
-			"password": "oneoneoneoneoneoneone111"
-		}`), &user2)
+		t.Log("SignIn")
 
-		require.Equal(t, "user2@dhis2.org", user2.Email)
-		require.Empty(t, user2.Password)
+		var tokens *token.Tokens
+		client.PostJSON(t, "/tokens", strings.NewReader(`{}`), &tokens, inttest.WithBasicAuth("user@dhis2.org", "oneoneoneoneoneoneone111"))
```

Comment:
Is the `strings.NewReader(`{}`)` needed? `nil` worked just fine before

Reply:
Please see the conversation we're having further up